Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -567,26 +567,26 @@ extension ProductStore {
 /// - invalidSKU: the SKU is invalid or duplicated.
 /// - unknown: other error cases.
 ///
-public enum ProductUpdateError: Error {
+public enum ProductUpdateError: Error, Equatable {
     case duplicatedSKU
     case invalidSKU
     case notFoundInStorage
-    case unknown
+    case unknown(error: AnyError)
 
     init(error: Error) {
         guard let dotcomError = error as? DotcomError else {
-            self = .unknown
+            self = .unknown(error: .init(error))
```

Comment:
A small improvement could be to add something like
```swift
extension Error {
   var toAnyError: AnyError {
       .init(self)
   }
}
```
To avoid this cumbersome `.init(error)` in multiple places

Reply:
I guess we could also make it conform to `LocalizedError` and have the best of both worlds!
```swift
extension AnyError: LocalizedError {
	public var errorDescription: String? {
		return error.localizedDescription
	}

	public var failureReason: String? {
		return (error as? LocalizedError)?.failureReason
	}

	public var helpAnchor: String? {
		return (error as? LocalizedError)?.helpAnchor
	}

	public var recoverySuggestion: String? {
		return (error as? LocalizedError)?.recoverySuggestion
	}
}
```

Reply:
cool suggestions! I improved them in https://github.com/woocommerce/woocommerce-ios/pull/2781/commits/79f536ed6dfbd1e0179f534bc3624c07114ae91a